Staging next that includes gcc8

FRidh opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

globin committed:
gcc: default to gcc8

Synthetica9 committed:
libomxil-bellagio: fix build on gcc8

Added the `-Wno-error=array-bounds` flag. Fixes #40213.

Synthetica9 committed:
systemd: suppress certain warnings to compile under gcc8

xeji committed:
Merge pull request #40246 from Synthetica9/gcc8-systemd-fix

systemd: suppress certain warnings to compile under gcc8

matthewbauer committed:
release: disable tests.cc-wrapper-gcc8.x86_64-darwin

This test is broken because gcc8.x86_64-darwin is broken. Please add
the test back in once we have fixed it!

Job: https://hydra.nixos.org/job/nixpkgs/trunk/gcc8.x86_64-darwin
Issue: #40038

Synthetica9 committed:
nasm: set -Wno-error=attributes for gcc8 (#40431)

Synthetica9 committed:
libglvnd: set -Wno-error=array-bounds (#40432)

Synthetica9 committed:
gcc: reset default to gcc7 on darwin

Synthetica9 committed:
gcc8: mark as broken on darwin

Synthetica9 committed:
gcc8: s/hostPlatform/stdenv/

Synthetica9 committed:
gcc8: disable test on darwin

matthewbauer committed:
Merge pull request #40519 from Synthetica9/gcc8-darwin-revert

gcc,gcc8: {revert gcc to gcc7, mark gcc8 as broken} on Darwin

Synthetica9 committed:
Merge remote-tracking branch 'upstream/master' into gcc8

Synthetica9 committed:
llvm6: default to gcc7 instead of gcc8

matthewbauer committed:
Merge pull request #40574 from Synthetica9/gcc8-master-merge

gcc8: merge branch master

matthewbauer committed:
Merge pull request #40592 from Synthetica9/gcc8-llvm6-revert

llvm6: default to gcc7 instead of gcc8

Synthetica9 committed:
iasl: 20180313 -> 20180508

Synthetica9 committed:
iasl: Temporary fix for acpica/acpica/#387

Synthetica9 committed:
dhcp: fix compilation on gcc8 (#40605)

xeji committed:
Merge pull request #40608 from Synthetica9/gcc8-iasl

iasl: 20180313 -> 20180508,  temporary fix for upstream issue #387

Synthetica9 committed:
hdf5: use gcc7 instead of gcc8 (#40704)

Synthetica9 committed:
elfutils: add patch that fixes compilation under gcc8 (#40705)

* elfutils: add patch that fixes compilation under gcc8

* elfutils: follow advice from @jtojnar

xeji committed:
Revert "iasl: 20180313 -> 20180508"

Broke xen and virtualbox.
This reverts commit d2f213216001232aa979f48b15a71999850635e0.

Synthetica9 committed:
iasl: integrate upstream patch for acpica/acpica#387 (#41481)

fpletz committed:
gcc: bump default to version 8

fpletz committed:
libglvnd: fix build with gcc8

fpletz committed:
libomxil-bellagio: fix build with gcc8

fpletz committed:
dhcp: fix build with gcc8

globin committed:
pfixtools: fix build with gcc8

fpletz committed:
blobfuse: fix build with gcc8

fpletz committed:
liboping: fix build with gcc8

fpletz committed:
getxbook: fix build with gcc8

globin committed:
llvm: add patch to fix build with gcc8

fpletz committed:
libsepol: 2.7 -> 2.8

fpletz committed:
libselinux: 2.7 -> 2.8

fpletz committed:
setools: 2017-11-10 -> 4.2.0, fix build with gcc8

fpletz committed:
telegram-cli: fix build with gcc8

globin committed:
gfortran8: init

fpletz committed:
openocd: fix build with gcc8

fpletz committed:
zookeeper_mt: fix build with gcc8

globin committed:
glibc: fix withGd build with gcc8

fpletz committed:
libextractor: fix build with gcc8

globin committed:
netsurf.libwapcaplet: fix build with gcc8

fpletz committed:
nginx: fix build with gcc8

globin committed:
libfaketime: fix build with gcc8

fpletz committed:
belle-sip: fix build with gcc8

globin committed:
libdynd: fix build with gcc8

globin committed:
vtk: 7.0.0 -> 7.1.1

globin committed:
bzrtp: fix build with gcc8

globin committed:
edk2: fix build with gcc8

fpletz committed:
gem-config: fix grpc build with gcc8

lheckemann committed:
aerospike: fix build with gcc8

globin committed:
mmc-utils: 2018-03-27 -> 2018-12-14

globin committed:
chez: fix build with gcc8

lheckemann committed:
altermime: fix build with gcc8

lheckemann committed:
bip: fix build with gcc8

fpletz committed:
xbps: fix build with gcc8

fpletz committed:
uri: 2017-07-16 -> 1.1.0, fix build with gcc8

fpletz committed:
leatherman: 1.5.3 -> 1.5.4, fix build with gcc8

fpletz committed:
libwhereami: 0.2.0 -> 0.2.2, fix build with gcc8

fpletz committed:
cpp-hocon: fix build with gcc8

fpletz committed:
telegram-purple: 1.3.0 -> 1.3.1, fix gcc8 build

globin committed:
libmaa: fix build with gcc8

fpletz committed:
facter: fix build with gcc8

fpletz committed:
zeroc_ice: fix build with gcc8

globin committed:
mediastreamer2: fix build with gcc8

fpletz committed:
papi: fix build with gcc8

globin committed:
mesos: fix build with gcc8

globin committed:
mps: fix build with gcc8

globin committed:
qpid-cpp: fix build with gcc8

fpletz committed:
iasl: 20181213 -> 20190108

fpletz committed:
smpeg: fix build with gcc8

fpletz committed:
agg: fix build with gcc8

fpletz committed:
chaps: fix build with gcc8

fpletz committed:
libselinux: use PYTHONLIBDIR instead of manual copy

fpletz committed:
ants: fix build with gcc8

fpletz committed:
gnuapl: fix build with gcc8

fpletz committed:
gnuclad: fix build with gcc8

fpletz committed:
darling-dmg: 1.0.4 -> 1.0.4+git20180914

marsam committed:
pythonPackages.area53: drop

PyPI package was taken offline.

See https://github.com/bluepines/slick53/issues/5

marsam committed:
pythonPackages.pystache: fix pname

marsam committed:
pythonPackages.pymysqlsa: fix PyPI name

marsam committed:
pythonPackages.pyaudio: fix PyPI name

marsam committed:
pythonPackages.mrbob: fix PyPI name

marsam committed:
pythonPackages.mp2: fix PyPI name

marsam committed:
pythonPackages.eggdeps: fix PyPI name

marsam committed:
pythonPackages.python3pika: drop

Pika already supports Python3, and python3pika wasn't updated since 2014

marsam committed:
pythonPackages.pyev: drop

pyev is not available in PyPI anymore

marsam committed:
pythonPackages.pyaudio: 0.2.9 -> 0.2.11

marsam committed:
pythonPackages.mpd2: 0.5.5 -> 1.0.0

fpletz committed:
linuxPackages.usbip: fix build with gcc8

dtzWill committed:
libX11: 1.6.7 -> 1.6.8

dtzWill committed:
libXi: 1.7.9 -> 1.7.10

dtzWill committed:
encodings: 1.0.4 -> 1.0.5

dtzWill committed:
sessreg: 1.1.1 -> 1.1.2

FRidh committed:
Split buildPythonPackage into setup hooks

This commit splits the `buildPythonPackage` into multiple setup hooks.

Generally, Python packages are built from source to wheels using `setuptools`.
The wheels are then installed with `pip`. Tests were often called with
`python setup.py test` but this is less common nowadays. Most projects
now use a different entry point for running tests, typically `pytest`
or `nosetests`.

Since the wheel format was introduced more tools were built to generate these,
e.g. `flit`. Since PEP 517 is provisionally accepted, defining a build-system
independent format (`pyproject.toml`), `pip` can now use that format to
execute the correct build-system.

In the past I've added support for PEP 517 (`pyproject`) to the Python
builder, resulting in a now rather large builder. Furthermore, it was not possible
to reuse components elsewhere. Therefore, the builder is now split into multiple
setup hooks.

The `setuptoolsCheckHook` is included now by default but in time it should
be removed from `buildPythonPackage` to make it easier to use another hook
(curently one has to pass in `dontUseSetuptoolsCheck`).

r-ryantm committed:
kexectools: 2.0.19 -> 2.0.20

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/kexec-tools/versions

dtzWill committed:
xauth: 1.0.10 -> 1.1

https://lists.x.org/archives/xorg-announce/2019-July/003005.html

teto committed:
linux: clearer origin for settings configuration

adds _file so that nix may have a chance to display what file the conflictings settings come
from.

tobim committed:
qt48: Fix build for gcc8

dtzWill committed:
nsd: 4.2.1 -> 4.2.2

dtzWill committed:
xorg.xinput: 1.6.2 -> 1.6.3

https://lists.x.org/archives/xorg-announce/2019-July/003013.html

dtzWill committed:
mesa_glu: 9.0.0 -> 9.0.1

* source tarball is now xz, not bz2
* pkgconfig was fixed to include Cflags

https://gitlab.freedesktop.org/mesa/glu/compare/glu-9.0.0...glu-9.0.1

globin committed:
Merge remote-tracking branch 'upstream/master' into gcc-8

globin committed:
getxbook: fix merge

globin committed:
rubygems.grpc: fix build with gcc8

globin committed:
arangodb_3_2: remove

doesn't support gcc 8

globin committed:
Merge pull request #67246 from tobim/gcc8/qt48

qt48: Fix build for gcc8

dtzWill committed:
libusb1: pname-ification

dtzWill committed:
libusb1: 1.0.22 -> 1.0.23

dtzWill committed:
webkitgtk: 2.24.3 -> 2.24.4 (security!)

https://webkitgtk.org/2019/08/28/webkitgtk2.24.4-released.html
https://webkitgtk.org/security/WSA-2019-0004.html

FRidh committed:
pythonPackages.tld: use pythonImportsCheck and pytestCheckHook

FRidh committed:
python.pkgs.terminado: disable tests

FRidh committed:
python: bootstrapped-pip: update wheel, setuptools, pip

wheel 33.4 -> 33.6
setuptools 41.0.1 -> 42.2.0
pip 19.1.1 -> 19.2.3

FRidh committed:
python.pkgs.pip: 19.1.1 -> 19.2.3

FRidh committed:
python.pkgs.setuptools: 0.33.4 -> 0.33.6

FRidh committed:
python.pkgs.wheel: 0.33.4 -> 0.33.6

jtojnar committed:
libgdata: 0.17.10 → 0.17.11

https://download.gnome.org/sources/libgdata/0.17/libgdata-0.17.11.news

jtojnar committed:
libgdata.tests: add installedTests

jtojnar committed:
glib-networking: format with nixpkgs-fmt

jtojnar committed:
glib-networking: drop unnecessary flag

auto_features are now enabled by default.

jtojnar committed:
glib-networking: improve meta

* Add maintainers
* Add homepage
* Correct license as per https://gitlab.gnome.org/GNOME/glib-networking/commit/b9c0324376a5e23b5f8210b0dd76858206baef26

jtojnar committed:
glib-networking: stop propagating dependencies

This was done from the beginning for no apparent reason:
https://github.com/NixOS/nixpkgs/commit/cc6ecaeeef96e4c09581b9bdb067269861b94c60

jtojnar committed:
glib-networking: clean up

jtojnar committed:
glib-networking: hardcode GSettings path to gnomeproxy module

So that this can be loaded from programs not depending on gsettings-desktop-schemas.

Currently, this patch is much uglier than it could be due to
https://gitlab.gnome.org/GNOME/glib/issues/1884

jtojnar committed:
glib-networking: add installed tests

jtojnar committed:
libproxy: remove outdated comment

We no longer propagate dependencies so the comment is not relevant.

jtojnar committed:
libproxy: split python outputs

jtojnar committed:
libproxy: format with nixpkgs-fmt

jtojnar committed:
libproxy: wrap GSetings helper

Add GSettings schemas required for GNOME helper.

dtzWill committed:
glib: 2.60.6 -> 2.60.7

https://gitlab.gnome.org/GNOME/glib/-/tags/2.60.7

dtzWill committed:
librsvg: 2.44.14 -> 2.44.15

https://mail.gnome.org/archives/ftp-release-list/2019-September/msg00029.html

dtzWill committed:
numactl: 2.0.12 -> 2.0.13

https://github.com/numactl/numactl/releases/tag/v2.0.13

dtzWill committed:
harfbuzz: 2.6.0 -> 2.6.1

https://github.com/harfbuzz/harfbuzz/releases/tag/2.6.1

dtzWill committed:
cups: 2.2.11 -> 2.2.12 (security!)

https://github.com/apple/cups/releases/tag/v2.2.12

worldofpeace committed:
Merge pull request #67957 from jtojnar/glib-networking-cleanup

glib-networking: clean up & add installed tests

worldofpeace committed:
Merge pull request #67939 from jtojnar/libgdata-0.17.11

libgdata: 0.17.10 → 0.17.11

fpletz committed:
llvm_4: fix build with gcc8

fpletz committed:
ocamlPackages.llvm: build with llvm_8

fpletz committed:
bap: build with llvm_8

fpletz committed:
irods: build with current version of llvm

fpletz committed:
llvm_38: drop since it doesn't build with gcc8

fpletz committed:
Revert "xbps: fix build with gcc8"

This reverts commit f0a1f7e02627e0e3f402ac55a7f6e112aba24df7.

fpletz committed:
Merge remote-tracking branch 'origin/master' into gcc-8

fpletz committed:
highlight: build with standard llvm version on darwin

dtzWill committed:
Merge pull request #66973 from dtzWill/update/xauth-1.1

xauth: 1.0.10 -> 1.1

orivej committed:
libjpeg-turbo: 2.0.2 -> 2.0.3

dtzWill committed:
Merge pull request #68074 from dtzWill/update/encodings-1.0.5

encodings: 1.0.4 -> 1.0.5

dtzWill committed:
Merge pull request #68052 from dtzWill/update/glib-2.60.7

glib: 2.60.6 -> 2.60.7

dtzWill committed:
Merge pull request #68070 from dtzWill/update/sessreg-1.1.2

sessreg: 1.1.1 -> 1.1.2

dtzWill committed:
Merge pull request #68054 from dtzWill/update/webkitgtk-2.24.4

webkitgtk: 2.24.3 -> 2.24.4 (security!)

mmahut committed:
Merge pull request #68079 from dtzWill/update/nsd-4.2.2

nsd: 4.2.1 -> 4.2.2

vcunat committed:
Merge branch 'staging-next' into staging

globin committed:
amarok: 2.9.0-20180618 -> 2.9.0-20190824

gcc 8 support

jonringer committed:
pythonPackages.setuptools_scm: 3.2.0 -> 3.3.3

jtojnar committed:
Merge branch 'staging-next' into staging

bhipple committed:
waf: 2.0.15 -> 2.0.18

FRidh committed:
pythonPackages.editorconfig: remove duplicate EditorConfig

FRidh committed:
pythonPackages.editorconfig: -> 0.12.1 -> 0.12.2

jtojnar committed:
Merge branch 'staging-next' into staging

FRidh committed:
Merge staging into staging-next

FRidh committed:
Merge remote-tracking branch 'upstream/gcc8' into staging-next

FRidh committed:
Revert "pythonPackages.pyev: drop"

Package is still referred to.

This reverts commit bb5c039fc95bea25133a01e8f1b46a50159f2412.

FRidh commented:
elfutils failure for aarach64 https://hydra.nixos.org/build/99978054

lheckemann committed:
Revert "elfutils: add patch that fixes compilation under gcc8 (#40705)"

This reverts commit b12448c001fd6f34099685662cb919752c84ab1f.

jonringer commented:
@GrahamcOfBorg eval

FRidh committed:
Merge master into staging-next

jtojnar committed:
Merge branch 'master' into staging-next

FRidh committed:
Merge master into staging-next

FRidh commented:
@vcunat can you cancel the staging-next builds of all but the last evaluation?

vcunat commented:
OK, I also triggered an evaluation, as I can see ~5k rebuilds per platform since the last one.

FRidh commented:
Thanks. Note python-unstable builds the exact same revision as staging-next now.

vcunat commented:
Oh, I looked wrong, I was one evaluation behind.  Anyway, only current version is being built now.

vcunat commented:
@grahamc any idea how to get Hydra job-management rights to @FRidh?

vcunat commented:
ghc on darwin shows a problem around sphinx, I suppose?
```
make -C utils/haddock/doc html SPHINX_BUILD=/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/bin/sphinx-build
/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/bin/sphinx-build -b html . .build-html
Traceback (most recent call last):
  File "/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/bin/.sphinx-build-wrapped", line 7, in <module>
    from sphinx.cmd.build import main
  File "/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/lib/python3.7/site-packages/sphinx/cmd/build.py", line 25, in <module>
    from sphinx.application import Sphinx
  File "/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/lib/python3.7/site-packages/sphinx/application.py", line 38, in <module>
    from sphinx.registry import SphinxComponentRegistry
  File "/nix/store/xd97694caj013pbf54mi3gnhk3jw5qga-python3.7-sphinx-1.8.3/lib/python3.7/site-packages/sphinx/registry.py", line 19, in <module>
    from pkg_resources import iter_entry_points
ModuleNotFoundError: No module named 'pkg_resources'
```

FRidh commented:
That's a regression from f7e28bf5d8181926e600a222cb70180519d09726. Looking at the changeset `setuptools` is no longer propagated and thus no longer added to the wrapper/injected. While this is desirable long-term, I did not intend to do this right now because this will cause more run-time issues. I can push a fix directly but this will cause a whole new rebuild again.

FRidh committed:
python.pkgs.sphinx: requires setuptools

sphinx has setuptools (pkg_resources) as a runtime-dependency

FRidh commented:
I've pushed a fix for specifically sphinx but not buildPython* in general. Depending on the amount of runtime failures we encounter we can always fix it there.

FRidh committed:
python.pkgs.httpretty: disable flaky test

vcunat commented:
Any idea whether this is still attempted for 19.09?

FRidh commented:
@disassembler mentioned on Discourse the branch-off was postponed till tomorrow so we can get this in. I think later tonight we should simply merge, despite a chance in regressions.

FRidh commented:
Jobs here can be cancelled https://hydra.nixos.org/jobset/nixpkgs/python-unstable

FRidh committed:
python.pkgs.pybind11: fix build

vcunat commented:
Cancelled.  `staging-next` is showing ~1k regressions against "trunk" already, though much of that might be some random failures.  Perhaps it depends whether "we" specifically want this change in 19.09 (or more precisely, whether the release managers do).

FRidh commented:
Qt 4 is broken on staging-next, that's causing the x86_64-linux failures. Aarch64 is also having more regressions.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/nixos-19-09-feature-freeze/3707/14

vcunat committed:
qt4: fixup build with gcc8 by Arch aur patch

I'll hope that this simple removal of "volatile" is OK.
I feel sure I saw a PR for this somewhere, but I can't find it.

FRidh commented:
@vcunat this PR for Qt48 and gcc8 ? https://github.com/NixOS/nixpkgs/pull/67246

vcunat commented:
That change was merged into gcc8 branch but I can't see it in staging-next :-/

FRidh commented:
Maybe gcc8 got force-pushed removing it

vcunat commented:
No, see comments on https://github.com/NixOS/nixpkgs/pull/63132

FRidh commented:
@vcunat will you merge gcc-8 or shall I?

vcunat commented:
I can't make it today.

FRidh committed:
Merge remote-tracking branch 'upstream/gcc-8' into staging-next

Earlier the gcc8 branch was merged instead of the gcc-8 branch (note the dash)...

FRidh committed:
Merge master into staging-next

FRidh commented:
OK. Now gcc-8 is in (5f4734b1dd8410d5287fe0f22b9c0f30850a2fc3).

andir commented:
Would you consider picking #68096 for this iteration of staging next as well? It seems to be fairly low impact on the overall systems besides the extra rebuilds. Having this through staging would help us having it accepted for 19.09.

vcunat commented:
Oh, I wondered if this release should only get one large feature merged on the last day (which is past the _planned_ fork-off date), but apparently we end up the same on every release :-)

FRidh committed:
Revert "qt4: fixup build with gcc8 by Arch aur patch"

The build was fixed in two ways, by both this patch and a substitution.
Let's keep the substitition instead of the patch as the patches may disappear.

This reverts commit ccc8c73ea00321318e9ac2cb75879f57bef328a7.

FRidh committed:
python.pkgs.tlsh: fix build

FRidh committed:
python2.pkgs.wxPython: fix build

FRidh committed:
gcc: use gcc7 for darwin

This was present in the gcc-8 branch but got removed in the merge.

FRidh commented:
proposed eval fix https://github.com/NixOS/nixpkgs/pull/68324

FRidh committed:
asn2quickder: fix build

nh2 commented:
It seems this merge broke building my builds with musl (due to a new `[-Werror=attribute-alias]` error appearing when `glibc` is being built with musl):

https://github.com/nh2/static-haskell-nix/issues/56

nh2 commented:
> It seems this merge broke building my builds with musl

PR to fix it: #71480

louisblin commented on pkgs/development/interpreters/python/mk-python-derivation.nix:
```diff
@@ -71,48 +92,61 @@ if disabled
   doCheck = false;
-  doInstallCheck = doCheck;
-  installCheckInputs = checkInputs;
+  doInstallCheck = attrs.doCheck or true;
```
After that change, when setting `doCheck = false` you get `doInstallCheck = false or true = true` and tests will still run. As far as I can tell, the python docs still recommend using `doCheck` to control whether tests run, so this looks like a regression.

vcunat replied:
Nitpick: Nix's "or" works for missing attributes, not for "falsy values".
